### DISCUSSION TITLE
Derivative field: Get Reference from reference

### DISCUSSION THREAD

**Discussion by webwurm** [AntlerVC/firetable#884] [Support / Q&A] [answered]
I have a problem, which I think shouldn't be too complicated to solve - but somehow I can't get it to work:

I would like to save the reference to a document, which is stored in a reference-field form a document, to which a current field points to. Sounds more complicated than it is:

1. current row; field "object_ref" points to a doc in the Objekte-collection
2. in Objekte-collection points a field "uid_user" to a doc in the Users-collection
^^ this reference I want in the current derivative field.

I tried this (besides many other things):
```
  const objectDoc = await db.collection("Objekte").doc(row.object_ref).get();
  return objectDoc.get('uid_user');
```
I also tried "toString()", but also that didn't work.

Can somebody help?

**Comment by BeeBombshell** [comment]
Hi @webwurm, I found a column in the [Connectors demo app table](https://demo.rowy.io/table/connectors#columnModal=%7B%22type%22%3A%22config%22%2C%22columnKey%22%3A%22collection%22%7D) that accesses data from another Firebase Collection and displays it as options. I hope this helps as a reference for what you're trying to achieve.

**Comment by webwurm** [comment, marked as answer]
I figured it out. It has to be a derivative-field with the type "Reference" and the following code:
```
  const objectDoc = await row.object_ref.get();
  return objectDoc.get('uid_user');
```

To get the field from a doc to which there is a reference in the current row is simply:
```
  const userDoc = await row.user_ref.get();
  return userDoc.get('email');
```
I was a bit confused by the "Intermediate example" [here]([url]https://docs.rowy.io/field-types/derivative(url)), but I actually have the reference in the field and not the id as in the example.
